### ISSUE TITLE
Editor freezes on first project load

### ISSUE THREAD

**Issue by phkeese** [godotengine/godot#47236]
**Godot version:**

v3.2.3.stable.official

**OS/device including version:**

MacBook Air 2020, macOS Big Sur
(Other machine) Windows 10

**Issue description:**

I have a project which freezes the editor on first load. I can copy the folder and verify the files to be *identical* and the editor will freeze the first time I try to load the copied project. I can change the main scene in `project.godot` and it will load fine.
Once the project is loaded, I can load it as often as I want, I can change the main scene back to the original, I can delete `.import` and even the editor config in `$HOME/Library/Application Support/Godot` and it will still load fine thereafter.

**Steps to reproduce:**
No idea, I noticed that the main scene is loaded the first time. It contains some fonts and stacked UI elements.
It also does not work if I remove dependencies of the scene. In that case it just shows the "Fix Dependencies" popup.

**Minimal reproduction project:**

[Project.zip](https://github.com/godotengine/godot/files/6177966/Project.zip)
This is the project, I have not been able to create a minimal version, so I am just submitting this one.
**Log**
```
Godot Engine v3.2.3.stable.official - https://godotengine.org
Using GLES2 video driver
OpenGL debugging not supported!
OpenGL ES 2.0 Renderer: Intel(R) Iris(TM) Plus Graphics OpenGL Engine
OpenGL ES 2.0 Batching: ON
	OPTIONS
	max_join_item_commands 16
	colored_vertex_format_threshold 0.25
	batch_buffer_size 16384
	light_scissor_area_threshold 1
	item_reordering_lookahead 4
	light_max_join_items 32
	single_rect_fallback False
	debug_flash False
	diagnose_frame False
CoreAudio: detected 2 channels
CoreAudio: audio buffer frames: 512 calculated latency: 11ms
 
Registered camera FaceTime HD Camera (Built-in) with id 1 position 0 at index 0
CORE API HASH: 17947557007960061372
EDITOR API HASH: 513236543915789544
Construct gdnative interface

Destruct gdnative interface

Class 'BulletPhysicsDirectBodyState' is not exposed, skipping.
Class 'BulletPhysicsDirectSpaceState' is not exposed, skipping.
Class 'BulletPhysicsServer' is not exposed, skipping.
Class 'GDScriptNativeClass' is not exposed, skipping.
Class 'IP_Unix' is not exposed, skipping.
Class 'InputDefault' is not exposed, skipping.
Class 'Physics2DDirectBodyStateSW' is not exposed, skipping.
Class 'Physics2DDirectSpaceStateSW' is not exposed, skipping.
Class 'Physics2DServerSW' is not exposed, skipping.
Class 'ResourceImporterOGGVorbis' is not exposed, skipping.
Loading resource: /Users/philipp/Library/Application Support/Godot/editor_settings-3.tres
EditorSettings: Load OK!
Loaded builtin certs
EditorSettings: Save OK!
Loading resource: res://pckrip/fonts/Kenney Future.ttf
Loading resource: res://pckrip/scenes/main.tscn
Loading resource: res://pckrip/scenes/paths.tscn
Loading resource: res://pckrip/scripts/paths.gd
Loading resource: res://pckrip/misc/theme.tres
Loading resource: res://pckrip/fonts/kenney-20.tres
Loading resource: res://pckrip/scenes/drop.tscn
Loading resource: res://pckrip/scripts/drop.gd
Loading resource: res://pckrip/fonts/kenney-50.tres
Loading resource: res://pckrip/scripts/main.gd
Loading resource: res://pckrip/scenes/extract.tscn
Loading resource: res://pckrip/fonts/kenney-12.tres
Loading resource: res://pckrip/scripts/extract.gd
Loading resource: res://pckrip/scripts/extractors/file_extractor.gd
Loading resource: res://pckrip/scripts/extractors/default.gd
Loading resource: res://pckrip/scripts/extractors/texture.gd
### This is where it gets stuck every time ###
```

**Debug Backtrace at moment of freeze**
```
thread #1, queue = 'com.apple.main-thread', stop reason = signal SIGSTOP
  * frame #0: 0x00007fff2033c0c6 libsystem_kernel.dylib`__psynch_mutexwait + 10
    frame #1: 0x00007fff2036c2c5 libsystem_pthread.dylib`_pthread_mutex_firstfit_lock_wait + 81
    frame #2: 0x00007fff2036a1bc libsystem_pthread.dylib`_pthread_mutex_firstfit_lock_slow + 211
    frame #3: 0x0000000101e1720c Godot`DynamicFontAtSize::get_char_size(wchar_t, wchar_t, Vector<Ref<DynamicFontAtSize> > const&) const + 140
    frame #4: 0x0000000101e1a40b Godot`DynamicFont::get_char_size(wchar_t, wchar_t) const + 43
    frame #5: 0x000000010181a0ad Godot`Label::get_longest_line_width() const + 141
    frame #6: 0x0000000101819632 Godot`Label::regenerate_word_cache() + 194
    frame #7: 0x0000000101819f3b Godot`Label::get_minimum_size() const + 171
    frame #8: 0x000000010177528e Godot`Control::get_combined_minimum_size() const + 30
    frame #9: 0x0000000101774317 Godot`Control::_size_changed() + 183
    frame #10: 0x0000000101738797 Godot`Label::_notificationv(int, bool) + 103
    frame #11: 0x00000001026cb523 Godot`Object::notification(int, bool) + 19
    frame #12: 0x00000001016b56a6 Godot`Node::_propagate_ready() + 134
    frame #13: 0x00000001016b568b Godot`Node::_propagate_ready() + 107
    frame #14: 0x00000001016b568b Godot`Node::_propagate_ready() + 107
    frame #15: 0x00000001016b568b Godot`Node::_propagate_ready() + 107
    frame #16: 0x00000001016bad14 Godot`Node::_add_child_nocheck(Node*, StringName const&) + 180
    frame #17: 0x0000000100df5739 Godot`EditorNode::load_scene(String const&, bool, bool, bool, bool) + 3705
    frame #18: 0x0000000100df43b1 Godot`EditorNode::_sources_changed(bool) + 113
    frame #19: 0x00000001026eeb09 Godot`MethodBind1<bool>::call(Object*, Variant const**, int, Variant::CallError&) + 281
    frame #20: 0x00000001026cea85 Godot`Object::call(StringName const&, Variant const**, int, Variant::CallError&) + 357
    frame #21: 0x00000001026d0e2d Godot`Object::emit_signal(StringName const&, Variant const**, int) + 1901
    frame #22: 0x00000001026cf292 Godot`Object::emit_signal(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) + 114
    frame #23: 0x0000000100d64bd4 Godot`EditorFileSystem::_notification(int) + 1636
    frame #24: 0x0000000100d743cc Godot`EditorFileSystem::_notificationv(int, bool) + 76
    frame #25: 0x00000001026cb523 Godot`Object::notification(int, bool) + 19
    frame #26: 0x00000001016ebdea Godot`SceneTree::_notify_group_pause(StringName const&, int) + 2074
    frame #27: 0x00000001016ec2d7 Godot`SceneTree::idle(float) + 599
    frame #28: 0x0000000100042a61 Godot`Main::iteration() + 817
    frame #29: 0x0000000100010457 Godot`OS_OSX::run() + 391
    frame #30: 0x00000001000164f1 Godot`main + 529
    frame #31: 0x00007fff20389621 libdyld.dylib`start + 1
    frame #32: 0x00007fff20389621 libdyld.dylib`start + 1
```

It seems like the call to `DynamicFontAtSize::get_char_size(wchar_t, wchar_t, Vector<Ref<DynamicFontAtSize> > const&)` causes a deadlock. There are several other threads waiting on semaphores (`_IP_ResolverPrivate::_thread_function(void*)`, `VisualServerScene::_gi_probe_bake_thread()`, `EditorExportPlatformAndroid::_check_for_changes_poll_thread(void*)` in combination with `OS_Unix::delay_usec(unsigned int)`, `CommandQueueMT::push_and_ret<VisualServerWrapMT, int (VisualServerWrapMT::*)(), int>(VisualServerWrapMT*, int (VisualServerWrapMT::*)(), int*)`)

Most of these are waiting on `OS_Unix::delay_usec(unsigned int)` or `SemaphoreOSX::wait()`.

**Comment by KoBeWi** [comment]
I have the same (or similar issue). No matter if I have scenes or scripts reopened or not, Godot will load everything and then do some additional process that isn't logged anywhere. In my case I have hundreds of files and it takes few minutes to load the project.

As for why it happens only on first load, I suspect it might be some file caching. I haven't tested but there's a chance that it doesn't happen on SSD.

**Comment by phkeese** [comment]
> I have the same (or similar issue). No matter if I have scenes or scripts reopened or not, Godot will load everything and then do some additional process that isn't logged anywhere. In my case I have hundreds of files and it takes few minutes to load the project.
> 
> As for why it happens only on first load, I suspect it might be some file caching. I haven't tested but there's a chance that it doesn't happen on SSD.

Both of my machines use SSDs, so I do not think that is the answer. Also, I have deleted the editor settings directory including the filesystem caches and it will still open the project fine. Does the editor write any other files, except for `$HOME/Library/Application Support/Godot` and the project directory?

**Comment by snougo** [comment]
I tried your project in 3.3 rc6 but didn't get any freeze issue
my device: 2013 late intel GPU macbook pro

**Comment by snougo** [comment]
> As for why it happens only on first load, I suspect it might be some file caching. I haven't tested but there's a chance that it doesn't happen on SSD.

that's right, when deleting the Cache Folder like on mac`/Users/username/Library/Caches/Godot`, Godot will recreat the cache files again that will cause a feeling of delay when you doing some opreations

**Comment by akien-mga** [comment]
This seems fixed in 3.3-stable, this was likely a duplicate of #42785. Please comment if you can still reproduce the issue.
